conbench/entities/benchmark_result.py

```diff
@@ -534,6 +522,22 @@ def unitsymbol(self) -> Optional[conbench.units.TUnit]:
 
         return conbench.units.legacy_convert(self.unit)
 
+    @property
+    def associated_commit_repo_url(self) -> str:
+        """
+        Always return a string. Return URL or "n/a".
+
+        This is for those consumers that absolutely need to have a string type
+        representation.
+        """
+        if self.commit and self.commit.repo_url is not None:
+            return self.commit.repo_url
+
+        # This means that the result is not associated with any commit, or it is
+        # associated with a legacy/invalid commit object in the database, one
+        # that does not have a repository URL set.
+        return "n/a"
```

Comment:
@jgehrcke As we talked about earlier, I rearranged things so that the diff only shows code that I changed. We can move code around later.

The only exception to this is this `associated_commit_repo_url ` property and some of the `Run.create()` internals, which are now in `BenchmarkResult.create()`.